checked/debug.go

```diff
@@ -114,24 +128,31 @@ type debugger struct {
 	entries [][]*debuggerEntry
 }
 
-func (d *debugger) append(event debuggerEvent, ref int, stack []byte) {
+func (d *debugger) append(event debuggerEvent, ref int, pc []uintptr) {
 	if len(d.entries) == 0 {
 		d.entries = make([][]*debuggerEntry, 1, tracebackCycles)
 	}
 	idx := len(d.entries) - 1
-	d.entries[idx] = append(d.entries[idx], &debuggerEntry{
-		event: event,
-		ref:   ref,
-		stack: stack,
-		t:     time.Now(),
-	})
+	entry := tracebackEntryPool.Get().(*debuggerEntry)
+	entry.event = event
+	entry.ref = ref
+	entry.pc = pc
+	entry.t = time.Now()
+	d.entries[idx] = append(d.entries[idx], entry)
 	if event == decRefEvent && ref == 0 {
 		if len(d.entries) == tracebackCycles {
 			// Shift all tracebacks back one if at end of traceback cycles
+			slice := d.entries[0]
+			for i, entry := range slice {
+				tracebackCallersPool.Put(entry.pc)
```

Comment:
I believe that the `interface{}` version of the `[]uintptr` retains the pointer to the still allocated array.  

Nil'ing it out below only releases the root from the `*debuggerEntry` to the `[]uintptr` which the pool still holds onto.